packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.tsx

```diff
@@ -15,13 +14,25 @@ import isShallowEqual from '@wordpress/is-shallow-equal';
  * Internal dependencies
  */
 import SlotFillContext from './slot-fill-context';
+import type {
+	SlotFillProviderProps,
+	BubblesVirtuallySlotFillContext,
+} from '../types';
 
-function createSlotRegistry() {
-	const slots = proxyMap();
-	const fills = proxyMap();
+function createSlotRegistry(): BubblesVirtuallySlotFillContext {
+	const slots: BubblesVirtuallySlotFillContext[ 'slots' ] = proxyMap();
+	const fills: BubblesVirtuallySlotFillContext[ 'fills' ] = proxyMap();
+
+	const registerSlot: BubblesVirtuallySlotFillContext[ 'registerSlot' ] = (
+		name,
+		ref,
+		fillProps
+	) => {
+		const slot = ( slots.get( name ) || {} ) as {
+			ref: typeof ref;
+			fillProps: typeof fillProps;
+		};
```

Comment:
I wonder if can help TS to better infer those types.

Reply:
How about something like this.

```
		const slot = slots.get( name ) || {
			ref: undefined,
			fillProps: undefined,
		};
```